Design/Mantid_As_a _SciPy_Styled_Package.md

```diff
@@ -0,0 +1,259 @@
+
+* Mantid.simpleapi
+    - An accumulation library that will import all of the algorithms within Mantid as it currently does
+* mantid.api
+    - the current mantid.api (workspaces, validators, algorithm)
+* mantid.io
+    - all loading and saving
+    - ?? should we split out NeXus, ISAW etc ??
+* mantid.math 
+    - matrix workspace math (also include `V3D`, `VMD`, ...)
+    - `Plus`, `ExponentialCorrection`, `Rebin`, `Fit`
+    - add aliases `Add`=`Plus`, `Subtract`=`Minus`
+* mantid.math.axes
+    - changing the axis of a matrix workspace (not the data), that is technique independent
+    - e.g. ScaleX`, `ConvertAxisByFormula`, `MedianBinWidth`
+* mantid.math.events
+    - deals with events (technique agnostic)
+    - `FilterEvents`, `RebinByPulseTime`
+* mantid.math.multidimensional
+    - technique agnostic multidimensional workspace operations
+    - does not include `ConvertToMD`
+* mantid.optimization
+    - Curve Fitting and Optimization
+    - e.g. Fit
+* mantid.instrument 
+    - grouping, masking, etc
+    - e.g. `GroupDetectors`, `MaskBTP`, `SetGoniometer`, `MoveInstrumentComponent`
+* mantid.workspace
+    - manipulate workspaces, history
+    - `RenameWorkspace`, `GroupWorkspaces`, `CompareWorkspaces`, `Comment`
+* mantid.metadata
+    - logs, titles, but not history
+* mantid.muons
+    - muon related algorithms and Fit Functions
+* mantid.neutrons
+    - things that are related to neutrons (time of flight), but not specific to a certain subfield (like diffraction)
+    - e.g. `ConvertUnits`, `ConvertToMD`, `NormaliseByCurrent`, `He3TubeEfficiency`
+* mantid.neutrons.crystal
+    - single crystal stuff. Will include `UnitCell`, `OrientedLattice`, `SymmetryOperation`
+    - e.g. `SetUB`, `FindPeaksMD`, `IndexPeaks`
+* mantid.neutrons.diffraction
+    - powder/amorphous diffraction stuff
+    - e.g. `StripVanadiumPeaks`, `AlignAndFocusPowder`
+* mantid.neutrons.inelastic
+    - algorithms related to both direct and indirect inelastic spectroscopy
+    - e.g. `GetEi`, `CorrectKiKf`, `SofQW`
+* mantid.neutrons.reactor
+    - single wavelength algorithms
+    - right now most are facility specific
+* mantid.neutrons.reflectometry
+    - e.g. `FindReflectometryLines`
+* mantid.neutrons.sans
+    - e.g. `CalculateEfficiency`
+* mantid.simulations
+    - deal with outside simulation programs (CASTEP, SASSENA, ...)
+    - `CalculateInelasticScatteringFromAbInitioPhonon` (`Abins`)
+* mantid.constants
+    - no algorithms here
+    - physical constants
+    - neutronic constants
+* mantid.remote
+    - `SubmitRemoteJob`, `AbortRemoteJob`
+
+**Facility Specific Libraries**
+
+* mantid.ess
+* mantid.hfir
+* mantid.ill
+* mantid.isis
+* mantid.sns
+* can add instrument or technique specific sublibraries
```

Comment:
I would suggest go further with this. In order not to to have highly facility specific things in say `mantid.neutrons.sans` I would suggest that Algorithms would live under `mantid.{facility}` until that have been proved to be useful/correct across sites. One common complaint that things in Mantid pretend to be general, when they are actually only implemented to work at one facility (possibly one instrument). Better to have more things in facility specific areas to reflect the reality.